From fc1e0732069e95f2de3cf1a22d15c44bbd3cfaae Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A9sz=C3=A1ros=20Mih=C3=A1ly?= <misi@majd.eu>
Date: Mon, 22 Jun 2020 00:08:12 +0200
Subject: [PATCH 1/2] init with zero any new or reused stun buffers

[Salvatore Bonaccorso: backport to 4.5.1.1: Use consistently ns_bzero as it
does not yet contain the upstream change 7a43aae7c3e1 ("Remove ns_bzero(),
ns_bcopy(), and ns_bcmp()"). Adjust for context changes.]
---
 src/apps/relay/ns_ioalib_engine_impl.c | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/src/apps/relay/ns_ioalib_engine_impl.c
+++ b/src/apps/relay/ns_ioalib_engine_impl.c
@@ -293,10 +293,19 @@ static stun_buffer_list_elem *new_blist_
 
 	if(!ret) {
 	  ret = (stun_buffer_list_elem *)turn_malloc(sizeof(stun_buffer_list_elem));
+	  /* init ns_bzero below will solve all of these in one step
 	  ret->buf.len = 0;
 	  ret->buf.offset = 0;
 	  ret->buf.coffset = 0;
+	  */
 	  ret->next = NULL;
+	  if (!ret) {
+		TURN_LOG_FUNC(TURN_LOG_LEVEL_ERROR, "%s: Cannot allocate memory for STUN buffer!\n", __FUNCTION__);
+	  }
+	}
+
+	if(ret) {
+	  ns_bzero(&ret->buf, sizeof(stun_buffer));
 	}
 
 	return ret;
